### ISSUE TITLE
translate-c: use modular arithmetic

### ISSUE THREAD

**Issue by alichraghi** [ziglang/zig#12465]
While writing a ZSTD binding, i got the following problem with translate-c.
C uses Wrapping Addition/Subtraction while operating with unsgined integers but Zig checks for integer overflow. which looks like a wrong behavior with translate-c.

e.g. Z Standard defines some error values like this:
```c
#define ZSTD_CONTENTSIZE_ERROR (0ULL - 2)
```
which will reproduce the below translate-c output, and throw integer overflow error:
```zig
pub const ZSTD_CONTENTSIZE_ERROR = @as(c_ulonglong, 0) - @as(c_int, 2);
```

Zig should detect if integer type is unsigned and then use Wrapping Addition/Subtraction.
```zig
pub const ZSTD_CONTENTSIZE_ERROR = @as(c_ulonglong, 0) -% @as(c_int, 2);
```

**Comment by ehaas** [comment]
Fixing this would require implementing the C implicit type promotion rules in the macro translator (which is not impossible but would be a significant amount of work). As a workaround, are you able to assign that constant to a `const unsigned long long` variable in C code that gets translated, and then use that constant? For example, in `my_zstd.h`:

```c
#include <zstd.h>
const unsigned long long MY_ZSTD_CONTENTSIZE_ERROR = ZSTD_CONTENTSIZE_ERROR;
```
This will work because `translate-c` will see the macro-expanded text, and it knows what the type of the result should be.

**Comment by alichraghi** [comment]
yeah that should work. for now i just copied the fix into [codebase](https://github.com/alichraghi/zstd.zig/blob/8412f20decd34af423bb5f5392f714408215a0e9/src/compress.zig#L8-L9)
